packages/nimble-components/src/banner/tests/banner-matrix.stories.ts

```diff
@@ -0,0 +1,100 @@
+import type { Story, Meta } from '@storybook/html';
+import { withXD } from 'storybook-addon-xd-designs';
+import { html, ViewTemplate, when } from '@microsoft/fast-element';
+import {
+    createMatrixThemeStory,
+    createStory
+} from '../../utilities/tests/storybook';
+import {
+    createMatrix,
+    sharedMatrixParameters
+} from '../../utilities/tests/matrix';
+import { hiddenWrapper } from '../../utilities/tests/hidden';
+import '../../all-components';
+import { loremIpsum } from '../../utilities/tests/lorem-ipsum';
+
+const metadata: Meta = {
+    title: 'Tests/Banner',
+    decorators: [withXD],
+    parameters: {
+        ...sharedMatrixParameters(),
+        design: {
+            artboardUrl:
+                'https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/29c405f7-08ea-48b6-973f-546970b9dbab/specs/'
+        }
+    }
+};
+
+export default metadata;
+
+const severityStates = [
+    ['Neutral', undefined],
+    ['Error', 'error'],
+    ['Warning', 'warning'],
+    ['Info', 'info']
+] as const;
+type SeverityState = typeof severityStates[number];
+
+const actionStates = [
+    ['', false, undefined],
+    ['link', true, undefined],
+    ['ghost button', false, 'ghost'],
+    ['outline button', false, 'outline']
+] as const;
+type ActionState = typeof actionStates[number];
+
+const partsHiddenStates = [
+    ['', false],
+    ['parts hidden', true]
+] as const;
+type PartsHiddenState = typeof partsHiddenStates[number];
+
+// prettier-ignore
+const component = (
+    [severityLabel, severity]: SeverityState,
+    [actionLabel, linkVisible, buttonAppearance]: ActionState,
+    [partsHiddenLabel, partsHidden]: PartsHiddenState
+): ViewTemplate => html`
+    <nimble-banner
+        open
+        severity="${severity!}"
```

Comment:
This `!` assertion seems wrong because `severity` can be undefined. Can you take a look at some of our other matrix stories to see how they address this?

Reply:
Looks like the answer is to use an arrow function, i.e. `() => severity`. Honestly, it seems odd that this avoids the problem, but it does.